src/dbus/interface/force_feedback.rs

```diff
@@ -0,0 +1,94 @@
+use std::{error::Error, future::Future};
+
+use packed_struct::types::{Integer, SizedInteger};
+use zbus::fdo;
+use zbus_macros::interface;
+
+use crate::{
+    drivers::steam_deck::hid_report::PackedRumbleReport,
+    input::{composite_device::client::CompositeDeviceClient, output_event::OutputEvent},
+};
+
+use super::NamedInterface;
+
+/// [ForceFeedbacker] is any device that can implement force feedback
+pub trait ForceFeedbacker {
+    fn rumble(&mut self, value: f64) -> impl Future<Output = Result<(), Box<dyn Error>>> + Send;
+    fn stop(&mut self) -> impl Future<Output = Result<(), Box<dyn Error>>> + Send;
+}
+
+impl ForceFeedbacker for CompositeDeviceClient {
+    async fn rumble(&mut self, value: f64) -> Result<(), Box<dyn Error>> {
+        let value = value.min(1.0);
+        let value = value.max(0.0);
+        log::debug!("Sending rumble event with value: {value}");
+        let report = PackedRumbleReport {
+            intensity: (value * u8::MAX as f64) as u8,
+            left_speed: Integer::from_primitive((value * u16::MAX as f64) as u16),
```

Comment:
This should be strong/weak magnitude. Steam deck does this in a non standard way and some evdev based devices implement it this way, but it's not a guarantee and is opinionated against standard.

Reply:
Not sure what you're asking here. You want this function to take two values; one for strong and weak magnitude?

Reply:
I think that is strange we're using the steam input rumble event as our "defacto" rumble when it's not a standard hid spec output event. This seems very opinionated towards the steam deck and steam input rather than being based on the hid spec. It probably doesn't matter in the long run technically because we translate between both of them which is why I didn't ask for changes, I just wanted to understand the reasoning.